Source/WebCore/xml/XPathPredicate.cpp

```diff
@@ -252,11 +252,16 @@ Union::Union(std::unique_ptr<Expression> lhs, std::unique_ptr<Expression> rhs)
 
 Value Union::evaluate() const
 {
+    EvaluationContext clonedContext(Expression::evaluationContext());
     Value lhsResult = subexpression(0).evaluate();
-    Value rhs = subexpression(1).evaluate();
+    Value rhsResult = [&] {
+        SetForScope<EvaluationContext> contextForScope(Expression::evaluationContext(), clonedContext);
```

Comment:
Not sure if I see why this needs to be scoped though.

Reply:
We want to replace `Expression::evaluationContext()` with `clonedContext` just for the evaluation of the `rhs`.
It is a little awkward but this is how we currently do it in XPathPredicate at the moment.

Blink passes the context as parameter to evaluate functions instead of having a global evaluationContext. I think this looks a bit nicer for cases like these so maybe we should refactor the same way.